### ISSUE TITLE
Remix doesn't split server code

### ISSUE THREAD

**Issue by jaschaio** [remix-run/remix#6318]
### What version of Remix are you using?

1.16.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Repository: https://github.com/jaschaio/remix-monorepo

**Manual steps to reproduce:**

Create a new directory and enter it:

```
mkdir -p remix-monorepo/packages
cd remix-monorepo/packages
```

Run `create-remix@latest`:

```
npx create-remix@latest
```

Choose  `app`, `Just the basics`,  `Remix App Server`, `TypeScript` and `Yes`:

<img width="934" alt="image" src="https://user-images.githubusercontent.com/11058666/236405665-f17f8bae-c883-4eca-be89-ef23f376dc5c.png">

Create another subdirectory within the `remix-monorepo/packages` folder with two files:

1. `remix-monorepo/packages/package.json`
2. `remix-monorepo/packages/User.ts` 

```
mkdir services && touch services/package.json && touch services/User.ts
```

Fill the `remix-monorepo/packages/package.json`:

```json
{
  "name": "services",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

Fill the `remix-monorepo/packages/User.ts`:

```js
// Any import from crypto will fail, e.G. createHmac, createDecipheriv or createHash
import { randomBytes } from 'crypto';

// Just pseudo code to make a point
export const getToken = ( length: number ) => {

    return randomBytes( Math.ceil( length / 2 ) )
        .toString( 'hex' )
        .slice( 0, length );

};
```

Add the `services` module as a dependency to the remix project

```
cd app && npm i ../services
```

Edit the `app/app/root.tsx` file:

```diff
import { cssBundleHref } from "@remix-run/css-bundle";
import type { LinksFunction } from "@remix-run/node";
import {
  Links,
  LiveReload,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from "@remix-run/react";

+ import { getToken } from 'services/User';

export const links: LinksFunction = () => [
  ...(cssBundleHref ? [{ rel: "stylesheet", href: cssBundleHref }] : []),
];

+export const loader = () => {
+
+  const token = getToken();
+
+  return {
+    token,
+  };

};

export default function App() {

+  console.log( useLoaderData );

  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width,initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}
```

Run `npm run dev`:

<img width="989" alt="image" src="https://user-images.githubusercontent.com/11058666/236408609-52ae7713-ae62-402f-9b21-5a2137ff61b3.png">

### Expected Behavior

Remix should strip server only code from client bundles and not throw any errors when trying to bundle node dependencies like `crypto`.

### Actual Behavior

Remix throws errors: 

<img width="989" alt="image" src="https://user-images.githubusercontent.com/11058666/236408609-52ae7713-ae62-402f-9b21-5a2137ff61b3.png">

I get it as well with other build in node modules:

<img width="843" alt="image" src="https://user-images.githubusercontent.com/11058666/236411339-37356379-010d-4ccf-a615-69c4967f9058.png">

Just a handful of things that throw errors for me:

```
import { randomBytes } from 'crypto';
import { writeFile } from 'node:fs';
import { Worker } from 'node:worker_threads';
import { PassThrough } from 'node:stream';
import { buffer } from 'node:stream/consumers';
```

**Comment by paul-vd** [comment]
It might be related: https://github.com/remix-run/remix/issues/6302

**Comment by pcattori** [comment]
Related to #6259

**Comment by joejemmely** [comment]
To get rid of the `The path "service/User" is imported...` warning, you'll want to update the `main` field in `remix-monorepo/packages/services
/package.json` to be `index.ts` (which you'll need to define and re-export from User). Alternatively you can configure exports using the `exports` fields if you want to exports from the modules directly.

Also, since you don't have a build step, you'll probably want Remix to bundle your service package. To do that you can configure `serverDependenciesToBundle` in your app `remix.config.js`

**Comment by jaschaio** [comment]
@joejemmely thanks, that get's indeed rid of the `The path "services/User" is imported...` warning but doesn't help with the `ERROR`.

Returning to `@remix-run/dev@1.13.0` resolves `npm run dev` for me, but `npm run build` continues to fail with a similar error:

<img width="646" alt="image" src="https://user-images.githubusercontent.com/11058666/236594508-e40b82e3-64d3-4f58-9e7d-286e1ccf854b.png">

**Comment by joejemmely** [comment]
Not a perfect solution but if you re-export the library from a `.server.ts` file it seems to work, check this example -> https://github.com/joejemmely/remix-mono/tree/main/services/remix-service/app/routes/a

**Comment by MichaelDeBoey** [comment]
We just published version `1.17.1-pre.0` which includes #6562 (a possible fix). If you'd like to take it for a test run please try it out and let us know what you think!
